README.md

```diff
@@ -272,38 +272,71 @@ KotlinPoet has classes for building each of these:
 import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
 
 val hoverboard = ClassName("com.mattel", "Hoverboard")
+val skateboard = ClassName("com.mattel", "Skateboard")
+val board = ClassName("com.mattel", "Board")
+
 val list = ClassName("kotlin.collections", "List")
 val arrayList = ClassName("kotlin.collections", "ArrayList")
 val listOfHoverboards = list.parameterizedBy(hoverboard)
 val arrayListOfHoverboards = arrayList.parameterizedBy(hoverboard)
+val listOfSkateboards = list.parameterizedBy(skateboard)
+
+val producerListOfBoards = list.parameterizedBy(WildcardTypeName.producerOf(board))
 
-val beyond = FunSpec.builder("beyond")
+val flyingThings = FunSpec.builder("flyingThings")
     .returns(listOfHoverboards)
     .addStatement("val result = %T()", arrayListOfHoverboards)
     .addStatement("result += %T()", hoverboard)
     .addStatement("result += %T()", hoverboard)
-    .addStatement("result += %T()", hoverboard)
     .addStatement("return result")
     .build()
+
+val notFlyingThings = FunSpec.builder("notFlyingThings")
+    .returns(listOfSkateboards)
+    .addStatement("return listOf(%T(), %T())", skateboard, skateboard)
+    .build()
+
+val things = FunSpec.builder("things")
+    .returns(producerListOfBoards)
+    .addParameter("thingFlies", ClassName("kotlin", "Boolean"))
+    .beginControlFlow("if(thingFlies) {")
```

Comment:
nit: space after `if` + you don't really need the closing bracket, `if (thingFlies)` will produce the same result